File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex <= 1:
            if rowIndex == 0:
                return [1]
            return [1, 1]
        
        prev_dp, dp = [1, 1], [1]
        
        for num in range(2, rowIndex + 1):
            dp = [1] * (num + 1)
            for i in range(1, num):
                dp[i] = prev_dp[i - 1] + prev_dp[i]
            prev_dp = dp
            
        return dp

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        if (rowIndex <= 1){
            if (rowIndex == 0)
                return vector<int>{1};
            return vector<int>{1, 1};
        }
        
        vector<int> prevDp{{1, 1}}, dp;
        
        for (int num = 2; num <= rowIndex; ++num){
            dp.assign(num + 1, 1);
            for (int i = 1; i < num; ++i)
                dp[i] = prevDp[i - 1] + prevDp[i];
            prevDp = dp;
        }
        
        return dp;
    }
};